crates/cargo-util-schemas/src/manifest/mod.rs

```diff
@@ -1503,14 +1503,29 @@ impl TomlLint {
             Self::Config(config) => config.priority,
         }
     }
+
+    pub fn sub_config(&self) -> Option<&TomlLintSubConfig> {
+        match self {
+            Self::Level(_) => None,
+            Self::Config(config) => config.sub_config.as_ref(),
+        }
+    }
 }
 
 #[derive(Serialize, Deserialize, Debug, Clone)]
 #[serde(rename_all = "kebab-case")]
 pub struct TomlLintConfig {
-    pub level: TomlLintLevel,
+    pub level: Option<TomlLintLevel>,
     #[serde(default)]
     pub priority: i8,
+    #[serde(flatten)]
+    pub sub_config: Option<TomlLintSubConfig>,
+}
+
+#[derive(Serialize, Deserialize, Debug, Clone)]
+#[serde(rename_all = "kebab-case")]
+pub enum TomlLintSubConfig {
+    CheckCfg(Vec<String>),
 }
```

Comment:
```suggestion
    #[serde(flatten)]
    pub config: toml::Table,
}
```
- this is lint config, so calling it `sub_config` is weird
- this design forces the every lint to have knowledge about every other lint's config.  Instead, we should treat these as unused keys

Reply:
It's now called `config` and is now as asked completely agnostic.